### DISCUSSION TITLE
How can I materialize a partition of an asset through the `dagster asset materialize` CLI command?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18114] [Q&A] [answered]
Hi, I want to use materialise an asset using the command `dagster asset materialize` with dagster cli <https://docs.dagster.io/_apidocs/cli>. How do I use this command with an asset that has partitions? It gives me error: `dagster._core.errors.DagsterInvariantViolationError: Cannot access partition_key for a non-partitioned run`

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700210764731179?thread_ts=1700210764.731179&cid=C01U954MEER).*

**Comment by cmpadden** [comment, marked as answer]
If you would like to materialize a specific partition, that can be done using the `--partition` flag. In the following example:
```python
# example-partitions.py

from dagster import asset, StaticPartitionsDefinition, Definitions

color_partitions = StaticPartitionsDefinition(["red", "yellow", "blue"])


@asset(partitions_def=color_partitions)
def color_asset(context):
    color = context.partition_key
    context.log.info(f"Processing color: {color}")
    return f"Data for color {color}"


defs = Definitions(assets=[color_asset])
```

One can use the following command to materialize the `red` partition:

```bash
$ dagster asset materialize -f example-partitions.py --select color_asset --partition red
```

**Reply by cmpadden** [reply]
Additionally, there is not currently a way to select multiple partitions / ranges. See the following issue to track this feature.

https://github.com/dagster-io/dagster/issues/21560
